
syntax = "proto3";

message Method {
	string clustering_name = 1; 
	string description = 2;
	// string member_type = 3;
	string clustering_method = 4;
	string clustering_method_parameters_JSON = 5; // maybe a stringified JSON object for this one.. hard to generalize to all clustering methods
	repeated string clustering_method_input_datatypes = 6; // something like iCluster uses multiple input types

}

message Cluster {
	string cluster_label = 1  //e.g. ic01, ic02
	repeated Method methods = 1
	repeated string members = 2
}

